crates/subspace-core-primitives/src/lib.rs

```diff
@@ -77,6 +77,22 @@ const PUBLIC_KEY_LENGTH: usize = 32;
 #[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
 pub struct PublicKey([u8; PUBLIC_KEY_LENGTH]);
 
+// TODO: add std::error::Error implementation
+/// Unexpected length for public key
+#[derive(Debug, Clone, Copy)]
+pub struct UnexpectedLengthError;
+
+impl PublicKey {
+    /// Converts slice to public key
+    pub fn from_slice(bytes: &[u8]) -> Result<Self, UnexpectedLengthError> {
+        bytes
+            .to_vec()
+            .try_into()
+            .map(From::<[u8; PUBLIC_KEY_LENGTH]>::from)
+            .map_err(|_| UnexpectedLengthError)
+    }
+}
```

Comment:
I'm wondering if we need this at all, I'd rather do this transformation in one place where we need it instead.

Reply:
Would it be better if I moved it in `From<schnorrkel::PublicKey>`, which is always infallible?

Reply:
Then you need to add `schnorrkel` as a dependency. `subspace-core-primitives` is intended to be a small and lightweight no-std compatible crate that we'll use in many places as well as will publish to crates.io for others to use since the data structures included are generally useful when dealing with the protocol.